### DISCUSSION TITLE
Path independent playlist formats?

### DISCUSSION THREAD

**Discussion by nichobi** [beetbox/beets#4032] [Q&A] [answered]
Does beets support any smarter and more stable ways of storing playlists? I currently use simple and handmade `m3u` files but a minor change from running `mbsync` or a change in path formatting usually leaves me with dozens of broken tracks.

Anyone have any solutions for this, either via beets or other software?

**Comment by sampsyo** [comment]
This would indeed be super cool, and it's an old idea for the beets data model that has not been implemented. For a sort of history of the options considered, you might see #123. Basically, I think a really elegant way to do this would be to essentially let tags be associated with an order—so a playlist would basically be a tag with ordering superpowers. But the devil's in the details, of course.

**Comment by RollingStar** [comment]
I haven't used it but I think Calliope is relevant.

https://discourse.beets.io/t/calliope-antisocial-music-recommendations/1532/

**Comment by nichobi** [comment, marked as answer]
I've been thinking about this for a long time, and ended up developing a plugin called [yapl](https://github.com/nichobi/beets-yapl) (**ya**ml **p**lay**l**ist), which defines playlists as yaml files (essentially just consisting of beet queries) and compiles them to m3u files.  Early days yet but it's solved my reliability concerns
